bitswap-monitoring-client/src/main.rs

```diff
@@ -224,14 +314,16 @@ async fn connect_and_receive(
                     monitor_name, origin_ip, origin_ma
                 );
 
-                let origin_country = match origin_ip {
+                let origin_country: celes::Country = match origin_ip {
                     None => prom::COUNTRY_NAME_UNKNOWN,
                     Some(ip) => match country_db.lookup::<maxminddb::geoip2::Country>(ip) {
                         Ok(country) => country
                             .country
                             .as_ref()
                             .map(|o| o.iso_code)
                             .flatten()
+                            .map(|iso_code| celes::Country::from_str(iso_code).ok())
```

Comment:
This costs performance. I'm pretty sure I designed the original to use alpha2 codes precisely to avoid doing this.

Reply:
Nur für mein Verständnis. Warum ist das hier so kostenintensiv? Wegen dem lowercase oder dem matchen?

In Zeile 274 folgende wird doch auch implizit kopiert weil wir den `origin_country` nicht borrowen können? Macht ein oder zweimal einen String kopieren wirklich soviel performance aus?

Reply:
Du warst einfach zu schnell :D 
Ich hatte schon verstanden das meine Frage nicht so schlau war und sie gelöscht. :D
Danke!